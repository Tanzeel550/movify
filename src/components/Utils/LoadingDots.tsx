import React from 'react';
import Loader from 'react-loader-spinner';

const LoadingDots: React.FC = () => (
  <div className="loading-dots">
    <Loader type="ThreeDots" color="#9400ff" height="100" width="100" />
  </div>
);

export default LoadingDots;
