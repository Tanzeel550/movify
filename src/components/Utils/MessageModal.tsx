import React from 'react';
import Modal from 'react-modal';

type props = {
  message: string;
  clearMessage: (params: any) => {};
  title: string;
};

const MessageModal = ({ message, clearMessage, title }: props) => (
  <Modal
    isOpen={!!message}
    onRequestClose={clearMessage}
    contentLabel="Modal"
    className="messageModal jumbotron"
    ariaHideApp={false}
  >
    <h3 className="messageModal__title">{title}</h3>
    <p className="messageModal__body">{message}</p>
    <button className="btn btn-light btn-lg" onClick={clearMessage}>
      Okay
    </button>
  </Modal>
);

export default MessageModal;
