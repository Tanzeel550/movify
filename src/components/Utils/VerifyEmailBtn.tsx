import React from 'react';
import { sendEmailVerification } from '../../actions/authActions';
import { connect, ConnectedProps } from 'react-redux';
import { setError } from '../../reducers/errorReducer';

const connector = connect(null, { setError });

type Props = ConnectedProps<typeof connector>;

export const VerifyEmailBtn: React.FC<Props> = ({ setError }: Props) => {
  return (
    <div>
      <button
        className="btn u-btn-colorized"
        onClick={async () => {
          try {
            await sendEmailVerification();
            throw new Error('A Verification Link has been sent to your email');
          } catch (e) {
            setError({ message: e.message, title: 'Verification Email' });
          }
        }}
      >
        Verify Email
      </button>
    </div>
  );
};

export default connector(VerifyEmailBtn);
