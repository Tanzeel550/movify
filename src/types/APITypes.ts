// ============================================================
// ======================== API Types =========================
// ============================================================

export type SEARCH_TYPE = {
  text: string;
};

export type APISingleMovieResultBySearch = {
  Title: string;
  Year: string;
  imdbID: string;
  Type: string;
  Poster: string;
};

export type APIMovieResultsBySearch = APISingleMovieResultBySearch[];

export type Rating = {
  Source: string;
  Value: string;
};

export type APIMovieResultByTitle = {
  Title?: string;
  Year?: string;
  Rated?: string;
  Released?: string;
  Runtime?: string;
  Genre?: string;
  Director?: string;
  Writer?: string;
  Actors?: string;
  Plot?: string;
  Language?: string;
  Country?: string;
  Awards?: string;
  Poster?: string;
  Ratings?: Rating[];
  Metascore?: string;
  imdbRating?: string;
  imdbVotes?: string;
  imdbID?: string;
  Type?: string;
  DVD?: string;
  BoxOffice?: string;
  Production?: string;
  Website?: string;
  Response?: string;
};

export type FireDBMovieItem = {
  id?: string;
  name: string;
  watched: boolean;
  dateWatched: any;
  whatYouLearnt: string;
  Poster: string;
  createdAt: number;
};
