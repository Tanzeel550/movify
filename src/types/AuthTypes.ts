// ============================================================
// ======================== Auth Types ========================
// ============================================================

export type AuthUserType = {
  user: object;
};

export type EmailPassType = {
  email: string;
  password: string;
};

export type AuthDefaultType = {
  isAuthenticated: boolean;
  user: object;
};

export type TestUserType = {
  uid: string;
  name: string;
  email: string;
  passion: string;
  os: string;
  version: string;
  height: number;
};
