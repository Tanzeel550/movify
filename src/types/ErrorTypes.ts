// ============================================================
// ======================= Error Types ========================
// ============================================================

export type ErrorState = {
  message: string;
  title: string;
};

export type ErrorActionsReturnType = {
  type: 'SET_ERROR' | 'CLEAR_ERROR';
  message?: string;
  title?: string;
};
