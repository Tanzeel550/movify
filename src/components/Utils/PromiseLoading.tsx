import React from 'react';
import { usePromiseTracker } from 'react-promise-tracker';
import LoadingDots from './LoadingDots';

type Props = {
  area: string;
};

const PromiseLoading = ({ area }: Props) => {
  const { promiseInProgress } = usePromiseTracker({ area });
  if (promiseInProgress) return <LoadingDots />;
  return <></>;
};

export default PromiseLoading;
