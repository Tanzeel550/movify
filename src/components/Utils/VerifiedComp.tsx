import React, { ReactElement } from 'react';
import VerifyEmailBtn from './VerifyEmailBtn';

type Props = {
  children:
    | React.ReactNode
    | ReactElement<any, any>
    | JSX.Element
    | JSX.Element[];
  // children: Component<{}, {}, any>;
  emailVerified: boolean;
};

const VerifiedComp: React.FC<Props> = ({ children, emailVerified }: Props) => {
  return emailVerified ? (
    <React.Fragment>{children}</React.Fragment>
  ) : (
    <div className="container jumbotron">
      <p>email is ${emailVerified}</p>
      <h3 className="u-text-colorized">
        You cannot access this page because your Email is not verified
      </h3>
      <VerifyEmailBtn />
    </div>
  );
};

export default VerifiedComp;
