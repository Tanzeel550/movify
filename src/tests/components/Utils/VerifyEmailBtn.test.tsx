import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { VerifyEmailBtn } from '../../../components/Utils/VerifyEmailBtn';
import { ActionCreatorWithPayload } from '@reduxjs/toolkit';
import { ErrorState } from '../../../types/ErrorTypes';

let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>,
  setError: ActionCreatorWithPayload<ErrorState, string> | jest.Mock<any, any>;

beforeEach(() => {
  setError = jest.fn();
  wrapper = shallow(<VerifyEmailBtn setError={setError} />);
});

test('should render VerifyEmailBtn correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

// TODO: convert sendEmailVerification in VerifyEmailBtn
// test('should call setError when button is pressed', () => {
//   wrapper.find('button').simulate('click');
//   expect(setError).toBeCalledTimes(1);
// });

export {};
