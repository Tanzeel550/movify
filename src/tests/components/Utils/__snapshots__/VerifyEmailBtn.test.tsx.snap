// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render VerifyEmailBtn correctly 1`] = `
<div>
  <button
    className="btn u-btn-colorized"
    onClick={[Function]}
  >
    Verify Email
  </button>
</div>
`;
